### DISCUSSION TITLE
[General] Would any gas fees incur on paymaster or user if paymaster returns `magic` value = 0 and transaction is failed on mainnet?

### DISCUSSION THREAD

**Discussion by hoshiyari420** [zkSync-Community-Hub/zksync-developers#601] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

I am trying to deploy paymaster that allows gas estimation instead of reverting by setting `magic` value to bytes4(0). 

While experimenting this behaviour on era_test_node(forking mainnet), it seemed that the transaction is halted, and the balance of paymaster is not decreased. 

Just want to ensure that this behaviour will be observed in future too, i.e. transaction will be halted in the validation phase itself if Magic bytes is set to 0 and no gas fees would incur on the paymaster.

**Comment by bxpana** [comment]
Thanks for the post. Will check with the team and get back to you

**Reply by bxpana** [reply, marked as answer]
Thanks for your patience, got an answer from the team. Since the validation has failed, the transaction is not even included into the blockchain
